src/targets/gpu/jit/mlir.cpp

```diff
@@ -171,9 +183,58 @@ struct mlir_compiler : compiler<mlir_compiler>
                                });
                 auto mlir =
                     insert_mlir(m, ins, any_cast<code_object_op>(ops[0]), dot_inputs_updated);
-                assert(contains(mods[1].inputs, split_ins));
-                auto pwm = mods[1];
-                pwm.replace(split_ins, mlir);
+                auto reshape_ins = mlir;
+                if(mods[1].mod.size() > 0)
+                {
+                    assert(contains(mods[1].inputs, split_ins));
+                    (void)(split_ins);
+                    assert(mods[1].mod.get_parameters().size() == 1);
+                    auto param_ins =
+                        std::find_if(mods[1].mod.begin(), mods[1].mod.end(), [](const auto& i) {
+                            return i.name() == "@param";
+                        });
+                    inputs_rep_map[param_ins] = mlir;
+                    reshape_ins =
+                        m.insert_instructions(
+                             ins,
+                             &mods[1].mod,
+                             &inputs_rep_map,
+                             [](module& insert_mod,
+                                instruction_ref insert_loc,
+                                const operation& op,
+                                const std::vector<instruction_ref>& inputs,
+                                const std::vector<module_ref>& mod_args) -> instruction_ref {
+                                 if(op.name() == "reshape")
+                                 {
+                                     // TODO: Add proper lowering for the reshape op
+                                     return insert_mod.insert_instruction(
```

Comment:
Should we check if the input is standard and then insert a contiguous?

Reply:
`reshape_lazy` can work some limited non-standard shapes too. 

Proper fix need to emulate what we do with `eliminate_contiguous`  that is try compute_shape() and if that throws an exception then only add `gpu::contiguous`. 

I added an `assert` for now